hedera-node/src/main/java/com/hedera/services/fees/charging/NarratedCharging.java

```diff
@@ -20,21 +20,27 @@
  * ‍
  */
 
-import com.hederahashgraph.api.proto.java.AccountID;
+import com.hedera.services.utils.TxnAccessor;
+import com.hederahashgraph.fee.FeeObject;
 
 /**
- * Defines a type able to determine if a given payer can afford
- * to pay a given amount.
- *
- * @author Michael Tinker
+ * Defines the checks and charging actions we need to apply the Services fee policy.
  */
-@FunctionalInterface
-public interface BalanceCheck {
-	/**
-	 * Flags if the given payer can afford the given amount.
-	 * @param payer a payer
-	 * @param amount an amount
-	 * @return if the payer can afford the amount
-	 */
-	boolean canAfford(AccountID payer, long amount);
+public interface NarratedCharging {
+	void setFees(FeeObject fees);
+	void resetForTxn(TxnAccessor accessor, long submittingNodeId);
+
+	boolean canPayerAffordAllFees();
+	boolean canPayerAffordNetworkFee();
+	boolean canPayerAffordServiceFee();
+	boolean isPayerWillingToCoverAllFees();
+	boolean isPayerWillingToCoverNetworkFee();
+	boolean isPayerWillingToCoverServiceFee();
+
+	void chargePayerAllFees();
+	void chargePayerServiceFee();
+	void chargePayerNetworkAndUpToNodeFee();
+	void chargeSubmittingNodeUpToNetworkFee();
```

Comment:
So this function will charge the submitting node up to the network fee as a penalty for not performing basic due diligence or for submitting a duplicate transaction with another node? I know some of the function names have enough details, but some of them may still need javadoc to explain.